lib/RoadizCoreBundle/src/Repository/NodeTypeFilesRepository.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RZ\Roadiz\CoreBundle\Repository;
+
+use RZ\Roadiz\CoreBundle\Entity\NodeType;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final readonly class NodeTypeFilesRepository implements NodeTypeRepositoryInterface
+{
+    public function __construct(
+        private string $nodesTypesDir,
+        private SerializerInterface $serializer,
+        private ValidatorInterface $validator,
+    ) {
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function findAll(): array
+    {
+        $finder = new Finder();
+        $finder->files()->in($this->nodesTypesDir);
+        if (!$finder->hasResults()) {
+            throw new \Exception('No files exist in this folder : '.$this->nodesTypesDir);
+        }
+        $nodeTypes = [];
+
+        foreach ($finder as $file) {
+            $content = $this->checkFile($file);
+            if (null === $content) {
+                continue;
+            }
+            $nodeTypes[] = $this->deserialize($content);
+        }
+
+        return $nodeTypes;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function findOneByName(string $name): ?NodeType
+    {
+        $finder = new Finder();
+        $finder->files()->in($this->nodesTypesDir);
+        if (!$finder->hasResults()) {
+            throw new \Exception('No files exist in this folder : '.$this->nodesTypesDir);
+        }
+
+        $finder->filter(function (\SplFileInfo $file) use ($name) {
+            return $this->supportName($file->getBasename(), $name);
+        });
+
+        $iterator = $finder->getIterator();
+        $iterator->rewind();
+        $firstFile = $iterator->current();
+
+        $content = $this->checkFile($firstFile);
+        if (null === $content) {
+            return null;
+        }
+
+        return $this->deserialize($content);
+    }
+
+    private function checkFile(?\SplFileInfo $file): ?string
+    {
+        if (null === $file) {
+            return null;
+        }
+        $content = file_get_contents($file->getRealPath());
+        if (false === $content) {
+            return null;
+        }
+        if (empty($content)) {
+            return null;
+        }
+
+        return $content;
+    }
+
+    private function supportName(string $fileName, string $name): bool
+    {
+        $supported = [
+            mb_strtolower($name),
+            mb_strtoupper($name),
```

Comment:
I think here you wanted to use ucfirst to capitalize only first letter, not the entire string.
We want to support `page.yml` , `Page.yml`, not `PAGE.yml`